Source/WebCore/platform/mediastream/RealtimeMediaSource.h

```diff
@@ -416,6 +421,25 @@ struct PhotoCapabilitiesOrError {
     String errorMessage;
 };
 
+struct PhotoSettingsOrError {
+    PhotoSettingsOrError() = default;
+    PhotoSettingsOrError(std::optional<PhotoSettings>&& settings, String&& errorMessage)
+        : settings(WTFMove(settings))
+        , errorMessage(WTFMove(errorMessage))
+    { }
+    PhotoSettingsOrError(PhotoSettings& settings)
+        : settings({ settings })
+    { }
+    explicit PhotoSettingsOrError(String&& errorMessage)
+        : errorMessage(WTFMove(errorMessage))
+    { }
+
+    operator bool() const { return settings.has_value(); }
```

Comment:
A typedef to Expected<PhotoSettings, String> would be a better fit rather than have a new container like this again.

Reply:
Also the String may not be moveable, would need to handle isolatedCopy() on it.

Reply:
This code path is all main thread only for now.